spec/spec.md

```diff
@@ -0,0 +1,2604 @@
+When analog signals are not given a concrete width, their widths are
+inferred according to a highly restrictive width inference rule, which
+requires that the widths of all arguments to a given attach operation be
+identical.
+
+    Analog<1>  ; 1-bit analog type
+    Analog<32> ; 32-bit analog type
+    Analog     ; analog type with inferred width
+
+## Vector Types
+
+A vector type is used to express an ordered sequence of elements of a
+given type. The length of the sequence must be non-negative and known.
+
+The following example specifies a ten element vector of 16-bit unsigned
+integers.
+
+    UInt<16>[10]
+
+The next example specifies a ten element vector of unsigned integers of
+omitted but identical bit widths.
+
+    UInt[10]
+
+Note that any type, including other aggregate types, may be used as the
+element type of the vector. The following example specifies a twenty
+element vector, each of which is a ten element vector of 16-bit unsigned
+integers.
+
+    UInt<16>[10][20]
+
+## Bundle Types
+
+A bundle type is used to express a collection of nested and named types.
+All fields in a bundle type must have a given name, and type.
+
+The following is an example of a possible type for representing a
+complex number. It has two fields, `real`, and `imag`, both 10-bit
+signed integers.
+
+    {real:SInt<10>, imag:SInt<10>}
+
+Additionally, a field may optionally be declared with a *flipped*
+orientation.
+
+    {word:UInt<32>, valid:UInt<1>, flip ready:UInt<1>}
+
+In a connection between circuit components with bundle types, the data
+carried by the flipped fields flow in the opposite direction as the data
+carried by the non-flipped fields.
+
+As an example, consider a module output port declared with the following
+type:
+
+    output a: {word:UInt<32>, valid:UInt<1>, flip ready:UInt<1>}
+
+In a connection to the `a` port, the data carried by the `word` and
+`valid` sub-fields will flow out of the module, while data carried by
+the `ready` sub-field will flow into the module. More details about how
+the bundle field orientation affects connections are explained in
+section [5.1](#connects).
+
+As in the case of vector types, a bundle field may be declared with any
+type, including other aggregate types.
+
+    {real: {word:UInt<32>, valid:UInt<1>, flip ready:UInt<1>}
+     imag: {word:UInt<32>, valid:UInt<1>, flip ready:UInt<1>}}
+
+When calculating the final direction of data flow, the orientation of a
+field is applied recursively to all nested types in the field. As an
+example, consider the following module port declared with a bundle type
+containing a nested bundle type.
+
+    output myport: {a: UInt, flip b: {c: UInt, flip d:UInt}}
+
+In a connection to `myport`, the `a` sub-field flows out of the module.
+The `c` sub-field contained in the `b` sub-field flows into the module,
+and the `d` sub-field contained in the `b` sub-field flows out of the
+module.
+
+## Passive Types {#passive_types}
```

Comment:
link to passive types is rendered

Reply:
@seldridge presumably you are OK with this? It is a bit awkward that the author now needs to know the exact section number and will have to manually update them in the text, and that the .md version has this shown which isn't searchable in the rendered version. I am OK with this but maybe worth noting in a "known and accepted issues" in the top of this PR description?

Reply:
There is a way to avoid this with pandoc where it'll auto-derive a cross-ref name without the user having to provide it.  E.g., this one should be `#passive-types` or `#passive_types` without the need for the `{#passive_types}`.  I believe I can then tell Pandoc to generate the cross-ref with some sigil, e.g., `#passive_types` -> `Section 5.4.1`.